client/js/widgets/widget.js

```diff
@@ -290,8 +290,14 @@ export class Widget extends StateManaged {
     }
   }
 
+  childClasses(child) {
```

Comment:
Not sure why this function is being added to all widgets.

Reply:
I did this so it could be called from line 300 regardless of the parent widget. If this isn't preferred, is the prefered style to check for the existence of a childClasses method on line 300? e.g.

```js
if(this.get('parent') && widgets.has(this.get('parent')) && widgets.get(this.get('parent')).childClasses)
  className += ' ' + widgets.get(this.get('parent')).childClasses(this);
```

Reply:
I see, children are asking any parent for classes that should be added but only piles are responding with classes to add.